src/core_types.jl

```diff
@@ -1,14 +1,20 @@
 """
-    abstract Rotation{T} <: StaticMatrix{T}
+    abstract Rotation{N,T} <: StaticMatrix{T}
 
-An abstract type representing 3D rotations. More abstractly, they represent
-3×3 unitary (orthogonal) matrices.
+An abstract type representing `N`-dimensional rotations. More abstractly, they represent
+unitary (orthogonal) `N`×`N` matrices.
 """
-abstract Rotation{T} <: StaticMatrix{T}
+abstract Rotation{N,T} <: StaticMatrix{T}
 
-Base.size{R <: Rotation}(::Union{R,Type{R}}) = (3,3)
+typealias Rotation2{T} Rotation{2,T}
+typealias Rotation3{T} Rotation{3,T}
+
+Base.size{N}(::Type{Rotation{N}}) = (N,N)
```

Comment:
IIUC you need `@pure` when you want some aspect of the code---like an `if` statement---to run at compile time. What could `@pure` change about this method?